modules/web/js/ballerina/diagram/views/compact/dimension-visitors/catch-statement-dimension-calculator-visitor.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import SizingUtil from './../sizing-util';
+
+/**
+ * Dimension visitor class for catch statement.
+ *
+ * @class CatchStatementDimensionCalculatorVisitor
+ * */
+class CatchStatementDimensionCalculatorVisitor {
```

Comment:
can we extend from a class like CompoundStatementDimensionCalculatorVisitor and have this as an empty class?

Reply:
This requires a major refactoring and will be focused in the future release